src/dmutils/sumo.py

```diff
@@ -0,0 +1,1289 @@
+WINDOW_DESCRIPTION_TO_SPEC = spark.createDataFrame(
+    [
+        ("Double, Clear, Thermal-Break, Air", 0.63, 0.62),
+        ("Double, Low-E, H-Gain", 0.29, 0.56),
+        ("Double, Low-E, L-Gain", 0.26, 0.31),
+        ("Single, Clear, Metal, Exterior Low-E Storm", 0.57, 0.47),
+        ("Single, Clear, Non-metal, Exterior Low-E Storm", 0.36, 0.46),
+        ("Double, Clear, Metal, Exterior Low-E Storm", 0.49, 0.44),
+        ("Double, Clear, Non-metal, Exterior Low-E Storm", 0.28, 0.42),
+        ("Double, Clear, Metal, Air", 0.76, 0.67),
+        ("Double, Clear, Metal, Air, Exterior Clear Storm", 0.55, 0.51),
+        ("Double, Clear, Non-metal, Air", 0.49, 0.56),
+        ("Double, Clear, Non-metal, Air, Exterior Clear Storm", 0.34, 0.49),
+        ("Double, Low-E, Non-metal, Air, M-Gain", 0.38, 0.44),
+        ("Double, Low-E, Non-metal, Air, L-Gain", 0.37, 0.30),
+        ("Single, Clear, Metal", 1.16, 0.76),
+        ("Single, Clear, Metal, Exterior Clear Storm", 0.67, 0.56),
+        ("Single, Clear, Non-metal", 0.84, 0.63),
+        ("Single, Clear, Non-metal, Exterior Clear Storm", 0.47, 0.54),
+        ("Triple, Low-E, Non-metal, Air, L-Gain", 0.29, 0.26),
+        ("Triple, Low-E, Insulated, Argon, H-Gain", 0.18, 0.40),
+        ("Triple, Low-E, Insulated, Argon, L-Gain", 0.17, 0.27),
+        ("No Windows", 0.84, 0.63),
+    ],
+    ("windows", "window_ufactor", "window_shgc"),
+)
+
+# list of columns containing fuel types for appliances
+APPLIANCE_FUEL_COLS = [
+    "clothes_dryer_fuel",
+    "cooking_range_fuel",
+    "heating_fuel",
+    "hot_tub_spa_fuel",
+    "pool_heater_fuel",
+    "water_heater_fuel",
+]
+
+# list of columns containing fuel types for appliances
+GAS_APPLIANCE_INDICATOR_COLS = [
+    "has_gas_fireplace",
+    "has_gas_grill",
+    "has_gas_lighting",
+]
+
+
+#  -- resstock reading and preprocessing functions  -- #
+
+
+def clean_building_metadata(raw_resstock_metadata_df: DataFrame) -> DataFrame:
+    """
+    Rename and remove columns of a ResStock building metadata dataframe
+
+    Can either pass a parquet file or an existing DataFrame.
+
+    Args:
+        raw_resstock_metadata_df (DataFrame): DataFrame containing raw ResStock metadata
+    Returns:
+        building_metadata_cleaned (DataFrame): cleaned ResStock building metadata
+
+    """
+    # Read in data and modify pkey name and dtype
+    building_metadata = raw_resstock_metadata_df.withColumn(
+        "building_id", F.col("bldg_id").cast("int")
+    ).drop("bldg_id")
+
+    # rename and remove columns
+    building_metadata_cleaned = data_cleaning.edit_columns(
+        df=building_metadata,
+        remove_substrings_from_columns=["in__"],
+        remove_columns_with_substrings=[
+            "simulation_control_run",
+            "emissions",
+            "weight",
+            "applicability",
+            "upgrade",
+        ],
+    )
+
+    return building_metadata_cleaned
+
+
+#  -- feature transformation udfs -- #
```

Comment:
Everything from here to L1164 is just copied straight from `build_feature_store.py`